ontobio/io/gpadparser.py

```diff
@@ -235,6 +239,8 @@ def parse_line(self, line):
         references = self.validate_curie_ids(assoc.evidence.has_supporting_reference, split_line)
```

Comment:
@mugitty Is there any overlap of logic between `self.validate_curie_ids` here and `self._validate_curie_using_db_xrefs` just below? Could `self._validate_curie_using_db_xrefs` be incorporated into `self.validate_curie_ids`?

Reply:
@dustine32 , validate_curie_ids calls _validate_id . This checks for things like "DB:id" or for annotations, prefix has to be in GO id space.  However, it does not validate against the syntax pattern specified in the db-xrefs file. This is meant as a catch-all for any identifier in the GAF line that has a database field and id_syntax

Reply:
@mugitty Would there be any issue if you incorporated this db-xrefs syntax pattern checking inside `_validate_id`? It [looks](https://github.com/biolink/ontobio/blob/3576a13929f01cd243b6401e2825405b5401e0b5/ontobio/io/assocparser.py#L830) like `_validate_id` assumes every `id` is a CURIE, which by definition should always have colon-separated database field and id_syntax (else an error will be reported) so both . The only complication I could think of is if a `metadata/db-xrefs` is not supplied when `_validate_id` is called but you could just make this optional in `_validate_id` (i.e., `if self.config.db_type_name_regex_id_syntax is not None`).

I guess my main concern is separating validation logic throughout the code that really should be in the same place.

Reply:
@dustine32, as you suggested, I could incorporate in _validate_id and add a check for self.config.db_type_name_regex_id_syntax is not None.  

Let me update